sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -424,6 +424,29 @@ class StringFunctionsSuite extends QueryTest with SharedSparkSession {
       df.selectExpr("substring_index(a, '.', 2)"),
       Row("www.apache")
     )
+
+    val testTable = "test_substring_index"
+    // count is int
+    withTempPath { path =>
+      withTable(testTable) {
+        sql(s"CREATE TABLE $testTable (num int) USING parquet LOCATION '${path.toURI.toString}'")
+        sql(s"INSERT INTO $testTable VALUES (1), (2), (3), (NULL)")
```

Comment:
Do we really need a NULL value? I think non-literal `count` parameter can trigger this bug.

Reply:
I think it makes sense to test `null` values. The relevant user discovered this problem through `null` values. In fact, when the `null` value is used as count, the result is `null` and no exception will be thrown.